SimDataFormats/Track/interface/SimTrack.h

```diff
@@ -44,12 +47,47 @@ class SimTrack : public CoreSimTrack {
 
   inline void setVertexIndex(const int v) { ivert = v; }
 
+  void setCrossedBoundaryVars(bool crossedBoundary,
+                              int idAtBoundary,
+                              math::XYZVectorD positionAtBoundary,
+                              math::XYZTLorentzVectorD momentumAtBoundary) {
+    crossedBoundary_ = crossedBoundary;
+    idAtBoundary_ = idAtBoundary;
+    positionAtBoundary_ = positionAtBoundary;
+    momentumAtBoundary_ = momentumAtBoundary;
+  }
+
+  bool crossedBoundary() const { return crossedBoundary_; }
+  math::XYZVectorD getPositionAtBoundary() const {
+    assertCrossedBoundary();
+    return positionAtBoundary_;
+  }
+  math::XYZTLorentzVectorD getMomentumAtBoundary() const {
+    assertCrossedBoundary();
+    return momentumAtBoundary_;
+  }
+  int getIDAtBoundary() const {
+    assertCrossedBoundary();
+    return idAtBoundary_;
+  }
+
 private:
   int ivert;
   int igenpart;
 
   math::XYZVectorD tkposition;
   math::XYZTLorentzVectorD tkmomentum;
+
+  bool crossedBoundary_;
+  int idAtBoundary_;
+  math::XYZVectorD positionAtBoundary_;
+  math::XYZTLorentzVectorD momentumAtBoundary_;
+
+  void assertCrossedBoundary() const {
```

Comment:
@tklijnsma , I would drop this method and this exception. Nothing dramatic happens if we do not know does track cross the boundary or not. The default should be "false" if code will manager to find out an intersection - set it to true.

Reply:
I've dropped the exception. I am somewhat worried about allowing unset variables to be returned without an exception though.